spec/power_helpers_spec.rb

```diff
@@ -2,12 +2,21 @@
   let(:response_body) { File.read('spec/fixtures/response.xml')  }
   let(:response) { Egauge::Response.new(response_body) }
   let(:url) { "http://foo.com" }
-  let(:expected_kwh) { { "solar" => 229, "solar_" => 230} }
+
   subject { Egauge::Client.new(url) }
 
-  before { allow(subject).to receive(:query).and_return(response)}
+  before { allow(subject).to receive(:query).and_return(response) }
 
-  it "should return a full day calculation of kwhs" do
-    expect(subject.full_day_kwh).to eq(expected_kwh)
+  describe '#monthly_kwh' do
+    let(:expected_query) {
+      {
+        :C => nil,
+        :T => "#{Time.now.to_i},1533096000,1530417600,1527825600,1525147200,1522555200,1519880400"
+      }
+    }
+    it "should return a full day calculation of kwhs" do
+      expect(subject).to receive(:query).with(expected_query)
+      subject.monthly_kwh(6)
```

Comment:
can you explain the math here or in the description maybe? I remember talking about it when you created the gem but memory failing me now...

Reply:
This spec in particular doesn't involve math. The only real math that takes place inside here is the conversion from Watt/sec to kWh, which is simply taking the value and dividing by 3,600,000.

Reply:
oh, i was confused. you're just checking that it recognizes 6 months in the query?